python-package/lightgbm/basic.py

```diff
@@ -2185,6 +2197,10 @@ def construct(self) -> "Dataset":
                         group_info = np.array(self.reference.group).astype(np.int32, copy=False)
                         _, self.group = np.unique(np.repeat(range(len(group_info)), repeats=group_info)[self.used_indices],
                                                   return_counts=True)
+                    if self.reference.position is not None:
+                        position_info = np.array(self.reference.position).astype(np.int32, copy=False)
+                        _, self.position = np.unique(np.repeat(range(len(position_info)), repeats=position_info)[self.used_indices],
+                                                     return_counts=True)
```

Comment:
```suggestion
```

This isn't necessary.

I suspect that maybe you just copied all code related to `group` in the Python packagge and replaced "group" with "position", without understanding what it did, so I'd like to explain why this is here to help your understanding of Python and LightGBM.

You don't see special handling of metadata like `weight`, `label`, and `init_score` here in this code constructing a subset because they have the same number of rows as the training data.

`group`, however, uses an approach similar to run length encoding

https://github.com/microsoft/LightGBM/blob/665c47313d6938a8d80559c824073a142c8bd870/python-package/lightgbm/basic.py#L1544-L1549

And so it has one element per group, not one per row in the training data.

Reply:
Sorry for the terrible mistake...